homeassistant/components/mochad.py

```diff
@@ -0,0 +1,78 @@
+"""
+Support for CM15A/CM19A X10 Controller using mochad daemon.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/mochad/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.const import (CONF_HOST, CONF_PORT)
+from homeassistant.helpers import config_validation as cv
+
+REQUIREMENTS = ['pymochad==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONTROLLER = None
+
+DOMAIN = 'mochad'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT): cv.port,
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Setup the mochad platform."""
+    conf = config[DOMAIN]
+    host = conf.get(CONF_HOST, 'localhost')
+    port = conf.get(CONF_PORT, '1099')
+    global CONTROLLER
+    CONTROLLER = MochadCtrl(host, port)
```

Comment:
If connection is not made, you can `return False`. This will prevent the platforms from getting loaded.